ReactiveXComponent/Parser/XCApiConfigParser.cs

```diff
@@ -11,55 +11,61 @@ namespace ReactiveXComponent.Parser
 {
     public class XCApiConfigParser : IXCApiConfigParser
     {
-        private XCApiDescription _xcApiDescription;
-
-        private Dictionary<string, int> _componentCodeByComponent;
-        private Dictionary<string, int> _stateMachineCodeByStateMachine;
+        private XCApiDescription _xcApiDescription;
+        private XNamespace _xc;
         private Dictionary<string, int> _eventCodeByEvent;
         private Dictionary<TopicIdentifier, string> _publisherTopicByIdentifier;
         private Dictionary<TopicIdentifier, string> _subscriberTopicByIdentifier;
         private Dictionary<long, string> _snapshotTopicByComponent;
+        private Dictionary<string, List<StateMachineInfo>> _stateMachineInfoByComponentRepository;
+        private Dictionary<string, int> _componentCodeByComponentNameRepo;
 
-        private XNamespace _xc;
 
         public void Parse(Stream xcApiStream)
         {
             var reader = XmlReader.Create(xcApiStream);
             _xcApiDescription = new XCApiDescription(reader);
             _xc = "http://xcomponent.com/DeploymentConfig.xsd";
-
-            _componentCodeByComponent = CreateComponentCodeByNameRepository(_xcApiDescription.GetComponentsNode());
-            _stateMachineCodeByStateMachine = CreateStateMachineCodeByNameRepository(_xcApiDescription.GetStateMachinesNode());
+            ParseComponentNodes(_xcApiDescription.GetComponentsNode(), out _componentCodeByComponentNameRepo, out _stateMachineInfoByComponentRepository);
```

Comment:
Remove out _componentCodeByComponentNameRepo, out _stateMachineInfoByComponentRepository and use them directly in the private method

Reply:
done